scripts/utils/environment.js

```diff
@@ -9,3 +14,76 @@ export const execOptions = { encoding: "utf8", stdio: "inherit" };
 export function configurePython(options = execOptions) {
   execSync("pipenv --python 3.8.17", options);
 }
+
+export function preExecuteChecks(pythonCheck = false, cookiecutterCheck = false) {
+  // Check if Node.js v18.x is installed
+  try {
+    invalid("Checking Node version");
+
+    const userNode = spawnSync("node", ["--version"], {
+      cwd: userdir,
+      shell: true,
+      encoding: "utf8"
+    });
+
+    const nvmrcPath = `${userdir}/.nvmrc`;
+    const systemNodeVersion = fs.readFileSync(nvmrcPath, "utf8").trim().split(".").slice(0, -2).join(".");
```

Comment:
This doesn't account for users that don't use nvm but are still on a valid Node version

Reply:
Or in other words, users that don't have this file in the `userdir`

Reply:
Ignore this, check this comment:
https://github.com/crowdbotics/modules/pull/1096#issuecomment-1944144800